### PR TITLE
Add derive builder

### PR DESCRIPTION
**Pull request by zancas**

### CODE REVIEW

File: zingolib/src/test_framework/mocks.rs [hunk 1]
```diff
@@ -73,58 +73,30 @@ pub fn random_zaddr() -> (
...
-            }
-        }
-
-        // Methods to set each field
-        build_method!(recipient, PaymentAddress);
-        build_method!(value, NoteValue);
-        build_method!(rseed, Rseed);
-
-        /// For any old zcaddr!
-        pub fn randomize_recipient(self) -> Self {
```
Comment (fluidvanadium) [inline zingolib/src/test_framework/mocks.rs]: this function may be necessary to distinguish notes in unit tests

File: zingolib/src/wallet/notes/sapling.rs [hunk 2]
```diff
@@ -189,104 +204,37 @@ impl ShieldedNoteInterface for SaplingNote {
 #[cfg(any(test, feature = "test-features"))]
 pub mod mocks {
```
Comment (fluidvanadium) [inline zingolib/src/wallet/notes/sapling.rs]: if this mod is empty, lets remove it

File: zingolib/src/wallet/notes/sapling.rs [hunk 3]
```diff
@@ -189,104 +204,37 @@ impl ShieldedNoteInterface for SaplingNote {
...
-    }
 }
 
 #[cfg(test)]
 mod tests {
-    use super::mocks::SaplingNoteBuilder;
+    use crate::test_framework::mocks::LRZSaplingNoteBuilder;
 
     #[test]
     pub fn build_sapling_note() {
```
Comment (fluidvanadium) [inline zingolib/src/wallet/notes/sapling.rs]: this test needs to build a SaplingNote, not just an LRZSaplingNote.

**Review by fluidvanadium — CHANGES_REQUESTED** [review]
its an improvement! it also needs some

File: zingolib/src/wallet/notes/sapling.rs (hunk 3 above)
Comment (zancas) [reply zingolib/src/wallet/notes/sapling.rs]: It does...   though.. I am not sure that a LRZSaplingNoteBuilder is still necessary.

Comment (zancas) [reply zingolib/src/wallet/notes/sapling.rs]: The `to_lrz` method converts to a Sapling Note.

File: zingolib/src/wallet/notes/sapling.rs (hunk 2 above)
Comment (zancas) [reply zingolib/src/wallet/notes/sapling.rs]: I think we'll want to reorganize mods a bit, perhaps as it's own PR (including removing this empty one).   I thought it should be it's own mod/name PR.

File: zingolib/src/wallet/notes/sapling.rs (hunk 3 above)
Comment (fluidvanadium) [reply zingolib/src/wallet/notes/sapling.rs]: the to_lrz method converts to a sapling_crypto::note

File: zingolib/src/test_framework/mocks.rs [hunk 4]
```diff
@@ -71,60 +70,32 @@ pub fn random_zaddr() -> (
 }
 
 // Sapling Note Mocker
-mod sapling_note {
+pub(crate) mod sapling_note {
```
Comment (fluidvanadium) [inline zingolib/src/test_framework/mocks.rs]: this mod should NOT be called sapling_note!!

File: zingolib/src/wallet/notes/sapling.rs (hunk 2 above)
Comment (fluidvanadium) [reply zingolib/src/wallet/notes/sapling.rs]: ok, open more PRs for these good ideas

File: zingolib/src/test_framework/mocks.rs (hunk 4 above)
Comment (fluidvanadium) [reply zingolib/src/test_framework/mocks.rs]: MY BAD!

File: zingolib/src/lightclient/deprecated.rs [hunk 5]
```diff
@@ -72,7 +72,7 @@ impl LightClient {
             .sapling_notes
             .iter()
             .filter(|nd| nd.is_change)
-            .map(|nd| nd.note.value().inner())
+            .map(|nd| nd.sapling_crypto_note.value().inner())
```
Comment (AloeareV) [inline zingolib/src/lightclient/deprecated.rs:75]: :smiley:

**Review by fluidvanadium — DISMISSED** [review]
I love #[derive(Builder

however, we cannot derive Builder on sapling_crypto::Note.
instead, i want us to set aside this portion of the changes and focus on landing add_derive_builder into point where it works best, which is zingolib/src/wallet/notes/sapling.rs SaplingNote

File: zingolib/src/wallet/notes/sapling.rs [hunk 6]
```diff
@@ -1,57 +1,75 @@
+#![allow(missing_docs)]
```
Comment (AloeareV) [inline zingolib/src/wallet/notes/sapling.rs]: I don't like this in the sapling mod. Why is this here? Is it the derive-generated SaplingNoteBuilder? Surely there's a way around putting this on the whole mod.

File: zingolib/src/wallet/notes/sapling.rs [hunk 7]
```diff
@@ -1,57 +1,75 @@
...
+use crate::wallet::{notes::ShieldedNoteInterface, traits::FromBytes};
+
 use super::{
     super::{data::TransactionRecord, Pool},
-    NoteInterface, ShieldedNoteInterface,
+    NoteInterface,
 };
 
 /// TODO: Add Doc Comment Here!
+#[derive(Builder)]
```
Comment (AloeareV) [inline zingolib/src/wallet/notes/sapling.rs:14]: I don't think this wants to exit in production code. Can we toss this behind a [cfg_attr](https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg_attr-attribute)?

**Review by AloeareV — CHANGES_REQUESTED** [review]

File: zingolib/src/wallet/notes/sapling.rs (hunk 6 above)
Comment (zancas) [reply zingolib/src/wallet/notes/sapling.rs]: I can shrink the scope.

File: zingolib/src/wallet/notes/sapling.rs (hunk 7 above)
Comment (zancas) [reply zingolib/src/wallet/notes/sapling.rs:14]: Yes.  It can.
Once that's done then I push the default impls into a feature-gated explicit impl block.
